### PR TITLE
[DROOLS-6586] fix xxe injection vulnerability

### PR DESCRIPTION
**Pull request by mariofusco**
**JIRA**: https://issues.redhat.com/browse/DROOLS-6586

### CODE REVIEW

**Comment by mariofusco** [comment]
jenkins execute fdb

File: drools-compiler/src/main/java/org/drools/compiler/kproject/models/KieModuleMarshaller.java [hunk 1]
```diff
@@ -185,12 +189,18 @@ private static void validate(String kModuleString) {
 
         private static void validate(Source source, Source duplicateSource) {
             try {
-                schema.newValidator().validate(source);
+                Validator validator = schema.newValidator();
+                validator.setProperty(XMLConstants.ACCESS_EXTERNAL_DTD, "");
+                validator.setProperty(XMLConstants.ACCESS_EXTERNAL_SCHEMA, "");
+                validator.validate(source);
             } catch (Exception schemaException) {
```
Comment (pibizza) [inline drools-compiler/src/main/java/org/drools/compiler/kproject/models/KieModuleMarshaller.java:182]: You can make a single method here and reduce lots of duplication.

Something like

Validator validator = createSecureValidatorFrom(schema);

private Validator createSecureValidatorFrom(Schema schema) {
                Validator validator = schema.newValidator();
                validator.setProperty(XMLConstants.ACCESS_EXTERNAL_DTD, "");
                validator.setProperty(XMLConstants.ACCESS_EXTERNAL_SCHEMA, "");
                return validator;
}

File: drools-compiler/src/main/java/org/drools/compiler/kproject/models/KieModuleMarshaller.java [hunk 2]
```diff
@@ -149,7 +152,8 @@ private static Schema loadSchema() {
         }
 
         private static Schema loadOldSchema() {
```
Comment (pibizza) [inline drools-compiler/src/main/java/org/drools/compiler/kproject/models/KieModuleMarshaller.java]: Can we make a single method with a parameter? It seems like there is a lot of shared code between loadSchema and loadOldSchema.

**Review by pibizza — CHANGES_REQUESTED** [review]

**Comment by mariofusco** [comment]
jenkins execute fdb

**Comment by mariofusco** [comment]
@pibizza duplication removed, thanks for suggestions.

**Review by hellowdan — APPROVED** [review]

**Review by pibizza — APPROVED** [review]
Looks good for me.

**Comment by mariofusco** [comment]
The build in failing because of the incompatibility of the `process-migration-service` with Java 8 but all kie-server tests and the whole droolsjbpm-integration are succeeding and in particular I see in the log

`23:01:44  [INFO] KIE :: Execution Server :: Tests :: Drools Integration Tests SUCCESS [20:17 min]`

**Review by lucamolteni — APPROVED** [review]

**Review by tarilabs — APPROVED** [review]
Yes, CDB is failing on multiple PRs for that PIM build and I concur unrelated 👍

**Comment by tarilabs** [comment]
(also for FDB, so I would merge it indeed)

**Pull request merged**